src/config.js

```diff
@@ -60,27 +58,36 @@ const h5pApiUrl = () => {
   }
 };
 
-module.exports = Object.assign(
-  {
-    componentName: process.env.npm_package_name,
-    host: process.env.EDITORIAL_FRONTEND_HOST || 'localhost',
-    port: process.env.EDITORIAL_FRONTEND_PORT || '3000',
-    redirectPort: process.env.NDLA_REDIRECT_PORT || '3001',
-    logEnvironment: process.env.NDLA_ENVIRONMENT || 'local',
-    logglyApiKey: process.env.LOGGLY_API_KEY,
-    ndlaApiUrl: process.env.NDLA_API_URL || apiDomain(),
-    editorialFrontendDomain: editorialFrontendDomain(),
-    learningpathFrontendDomain: learningpathFrontendDomain(),
-    ndlaPersonalClientId: process.env.NDLA_PERSONAL_CLIENT_ID || '',
-    auth0Domain: process.env.AUTH0_DOMAIN || '',
-    brightCoveAccountId: process.env.BRIGHTCOVE_ACCOUNT_ID || '123456789',
-    brightcovePlayerId: process.env.BRIGHTCOVE_PLAYER_ID || 'Ab1234',
-    brightcoveApiUrl: 'https://cms.api.brightcove.com',
-    h5pApiUrl: process.env.H5P_API_URL || h5pApiUrl(),
-    googleSearchApiUrl:
-      process.env.NDLA_GOOGLE_API_URL || 'https://www.googleapis.com',
-    googleSearchApiKey: process.env.NDLA_GOOGLE_API_KEY,
-    googleSearchEngineId: process.env.NDLA_GOOGLE_SEARCH_ENGINE_ID,
-  },
-  environment,
-);
+const config = {
+  componentName: getEnvironmentVariabel('npm_package_name'),
+  host: getEnvironmentVariabel('EDITORIAL_FRONTEND_HOST', 'localhost'),
+  port: getEnvironmentVariabel('EDITORIAL_FRONTEND_PORT', '3000'),
+  redirectPort: getEnvironmentVariabel('NDLA_REDIRECT_PORT', '3001'),
+  logEnvironment: getEnvironmentVariabel('NDLA_ENVIRONMENT', 'local'),
+  logglyApiKey: getEnvironmentVariabel('LOGGLY_API_KEY'),
+  isNdlaProdEnvironment: ndlaEnvironment === 'prod',
+  ndlaApiUrl: getEnvironmentVariabel('NDLA_API_URL', apiDomain()),
+  editorialFrontendDomain: editorialFrontendDomain(),
+  learningpathFrontendDomain: learningpathFrontendDomain(),
+  ndlaPersonalClientId: getEnvironmentVariabel('NDLA_PERSONAL_CLIENT_ID', ''),
+  auth0Domain: getEnvironmentVariabel('AUTH0_DOMAIN', ''),
+  brightCoveAccountId: getEnvironmentVariabel(
+    'BRIGHTCOVE_ACCOUNT_ID',
+    '123456789',
+  ),
+  brightcovePlayerId: getEnvironmentVariabel('BRIGHTCOVE_PLAYER_ID', 'Ab1234'),
+  brightcoveApiUrl: 'https://cms.api.brightcove.com',
+  h5pApiUrl: getEnvironmentVariabel('H5P_API_URL', h5pApiUrl()),
+  googleSearchApiUrl: getEnvironmentVariabel(
+    'NDLA_GOOGLE_API_URL',
+    'https://www.googleapis.com',
+  ),
+  googleSearchApiKey: getEnvironmentVariabel('NDLA_GOOGLE_API_KEY'),
+  googleSearchEngineId: getEnvironmentVariabel('NDLA_GOOGLE_SEARCH_ENGINE_ID'),
+};
+
+export function getUniversalConfig() {
+  return process.env.NODE_ENV === 'unittest' ? config : window.config;
```

Comment:
Should also have `process.env.BUILD_TARGET === 'server' ` check? For `Html.js` file where you add config to window.